lib/geoengineer/resources/aws/iam/aws_iam_saml_provider.rb

```diff
@@ -6,16 +6,23 @@
 class GeoEngineer::Resources::AwsIamSamlProvider < GeoEngineer::Resource
   validate -> { validate_required_attributes([:name, :saml_metadata_document]) }
 
-  after :initialize, -> { _terraform_id -> { name } }
+  after :initialize, -> { _terraform_id -> { NullObject.maybe(remote_resource)._terraform_id } }
+  after :initialize, -> { _geo_id -> { name.to_s } }
+
+  def support_tags?
+    false
+  end
 
   def self._fetch_remote_resources(provider)
+    idp_reg = %r{^arn:aws:iam::[0-9]{12}:saml-provider/(.+)$}
     AwsClients.iam(provider)
               .list_saml_providers
               .saml_provider_list
               .map do |idp|
+                name = idp.arn[idp_reg, 1]
```

Comment:
Unfortunately, the idp does not have the name field on it, only the arn, so I must extract it with a regex.

https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/IAM/Types/SAMLProviderListEntry.html